simespresso/io_QE/espresso_class.py

```diff
@@ -0,0 +1,815 @@
+                logging.debug('line in atomic positions section:' + str(line))
+                values = line.split()
+                print('values:' + str(values))
+                atom_pos = [0, 0, 0]
+                if values[0] in atomtypes:
+                    atomtype = values[0]
+                    # store position in meters; original in Angstrom
+                    atom_pos[0] = float(values[1]) * 1e-10
+                    atom_pos[1] = float(values[2]) * 1e-10
+                    atom_pos[2] = float(values[3]) * 1e-10
+                    # s = str(i)
+                    # make uid using string of index
+                    # uidstring =
+                    # uid = uuid.UUID(uidstring)
+
+                    p = Particle([atom_pos[0], atom_pos[1], atom_pos[2]])
+                    # ,uuid.UUID(int=i)
+                    print('uid:' + str(p.uid))
+                    p.data[CUBA.CHEMICAL_SPECIE] = atomtype
+                    particle_list.append(p)
+                    try:
+                        line = f.next()
+                    except StopIteration:
+                        print('EOF')
+                        break
+            if len(particle_list):
+                pc.add_particles(particle_list)
+                #  part_container.add_particle(p)
+                n = self.count_particles()
+                print('n_particles=' + str(n))
+        except StopIteration:
+            return ('EOF')
+
+        return pc
+
+    def count_particles(self):
+        n = 0
+        for particle in self.pc.iter_particles():
+            n += 1
+        return n
+
+    def start_qe(self, name_in, name_out, path_to_espresso='./pw.x',
+                 mpi=False, mpi_Nprocessors=2):
+        print('starting start_qe')
+        if path_to_espresso is None:
+            path_to_espresso = './pw.x '
+# I'll assume if no path given then pw.x is on path
+        if mpi:
+            command = 'mpirun -np ' + str(mpi_Nprocessors) + ' ' + \
+                      path_to_espresso + ' < ' + name_in + ' > ' + name_out
+        else:
+            command = path_to_espresso + ' < ' + name_in + ' > ' + name_out
+
+#        if not os.path.isfile(path_to_espresso):
+#            logging.warning(path_to_espresso + ' is not on path')
+#           this may be ok if pw.x is defined somewhere on the PATH
+#            return None
+
+        print('start_qe attempting to run: ' + command)
+# alternative would be to use subprocess.check_call()  -
+# however this would give the same info as the
+# try/except, while taking twice as long in the case of success, iiuc
+        try:
+            subprocess.check_call(
+                command, shell=True,
+                stdout=subprocess.PIPE).stdout.read()
+
+#            subprocess.Popen(command, shell=True,
+#                             stdout=subprocess.PIPE).stdout.read()
+        except:
+            e = sys.exc_info()[0]
+            print("<p>Error: %s</p>" % e)
```

Comment:
On travis, QE is not installed (yet) so it should fail at the moment.  Instead, just some information is printed  (see https://travis-ci.org/simphony/simphony-espresso-dft/jobs/95229232_L515)and the tests pass.  This should be changed.